test/lib/failOnError.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+var assert = require('assert');
+var AwsHelper = require('./../../lib/index');
+var simple = require('simple-mock');
+
+describe('AwsHelper.failOnError', function (done) {
+  afterEach(function () {
+    simple.restore();
+  });
+
+  it('should do nothing if there is no error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var spy = simple.spy(fakeContext, 'fail');
+    AwsHelper.failOnError(null, {}, fakeContext);
+    assert(!spy.called);
+    done();
+  });
+
+  it('should call context fail function if error', function (done) {
+    var fakeContext = {
+      fail: function () {}
+    };
+    var fakeError = 'There is an error';
+    var failMock = simple.mock(fakeContext, 'fail');
```

Comment:
@Kumjami do you not feel this is clearer to the [`next guy`](http://blog.codinghorror.com/coding-for-violent-psychopaths/) ? :wink: